File: app/src/main/java/opencrowd/hgc/hgcwallet/ui/main/developertool/CreateNodeScreen.java
package opencrowd.hgc.hgcwallet.ui.main.developertool;


import android.content.Context;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;

import com.wealthfront.magellan.BaseScreenView;
import com.wealthfront.magellan.Screen;

import opencrowd.hgc.hgcwallet.R;
import opencrowd.hgc.hgcwallet.database.DBHelper;
import opencrowd.hgc.hgcwallet.database.node.Node;
import opencrowd.hgc.hgcwallet.modals.HGCAccountID;
import opencrowd.hgc.hgcwallet.ui.customviews.TitleBarWrapper;

public class CreateNodeScreen extends Screen<CreateNodeScreenView> {

    Node node;
    public CreateNodeScreen(Node node) {
        this.node = node;
    }
    @Override
    protected CreateNodeScreenView createView(Context context) {
        return new CreateNodeScreenView(context,node);
    }
}

class CreateNodeScreenView extends BaseScreenView<CreateNodeScreen> {

    private EditText mEditTextHost,mEditTextPort,mEditTextAccountId;
    private CheckBox mCheckBoxIsActive;
    private Button mButtonsave;
    public CreateNodeScreenView(Context context, final Node paramNode) {
        super(context);
        inflate(context, R.layout.view_node_add_layout,this);
        TitleBarWrapper titleBar = new TitleBarWrapper(findViewById(R.id.titleBar));
        titleBar.setTitle(paramNode == null ? "Add Node" : "Edit Node");
        mEditTextHost = (EditText)findViewById(R.id.edittext_node_host);
        mEditTextPort = (EditText)findViewById(R.id.edittext_node_port);
        mEditTextAccountId = (EditText)findViewById(R.id.edittext_node_account);
        mCheckBoxIsActive = (CheckBox)findViewById(R.id.checkBox_active);
        mButtonsave = (Button)findViewById(R.id.btn_save);

        if(paramNode != null) {
            if(paramNode.getHost() != null) {
                mEditTextHost.setText(paramNode.getHost());
            }
            mEditTextPort.setText(String.valueOf(paramNode.getPort()));
            HGCAccountID accountID = paramNode.accountID();
            mEditTextAccountId.setText(accountID.stringRepresentation());
            mCheckBoxIsActive.setChecked(!paramNode.getDisabled());
        }

        mButtonsave.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                Node node = paramNode;
                if (paramNode == null) {
                    node = new Node();
                }
                if(!mEditTextHost.getText().toString().isEmpty()) {
                    node.setHost( mEditTextHost.getText().toString());
                }

                if(!mEditTextPort.getText().toString().isEmpty()) {
                    node.setPort(Integer.parseInt(mEditTextPort.getText().toString()));
                }

                if(!mEditTextAccountId.getText().toString().isEmpty()) {
                    HGCAccountID accountID = HGCAccountID.fromString(mEditTextAccountId.getText().toString());
                    if(accountID != null) {
                        node.setAccountID(accountID);
                    }
                }
                node.setDisabled( !mCheckBoxIsActive.isChecked());

                if (paramNode == null) {
                    DBHelper.createNode(node);

                } else  {
                    DBHelper.updateNode(node);

                }
                getScreen().getNavigator().goBack();

            }
        });
    }
}

File: app/src/main/java/hedera/hgc/hgcwallet/ui/main/developertool/CreateNodeScreen.kt
package hedera.hgc.hgcwallet.ui.main.developertool


import android.content.Context
import android.view.View
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText

import com.wealthfront.magellan.BaseScreenView
import com.wealthfront.magellan.Screen

import hedera.hgc.hgcwallet.R
import hedera.hgc.hgcwallet.database.DBHelper
import hedera.hgc.hgcwallet.database.node.Node
import hedera.hgc.hgcwallet.modals.HGCAccountID
import hedera.hgc.hgcwallet.ui.customviews.TitleBarWrapper

class CreateNodeScreen(node: Node?) : Screen<CreateNodeScreenView>() {

    data class Params(val node: Node?)

    private val params = Params(node)
    override fun createView(context: Context): CreateNodeScreenView {
        return CreateNodeScreenView(context, params)
    }

    internal fun onSaveClick(host: String, port: String, accountID: String, isDisabled: Boolean) {
        val node = params.node ?: Node()

        if (host.isNotEmpty())
            node.host = host

        if (port.isNotEmpty())
            node.port = Integer.parseInt(port)

        if (accountID.isNotEmpty()) {
            HGCAccountID.fromString(accountID)?.let {
                node.setAccountID(it)
            }
        }
        node.disabled = isDisabled

        if (params.node == null) DBHelper.createNode(node) else DBHelper.updateNode(node)

        navigator?.goBack()
    }
}

class CreateNodeScreenView(context: Context, val params: CreateNodeScreen.Params) : BaseScreenView<CreateNodeScreen>(context) {

    private val mEditTextHost: EditText
    private val mEditTextPort: EditText
    private val mEditTextAccountId: EditText
    private val mCheckBoxIsActive: CheckBox
    private val mButtonsave: Button
    private val titleBar: TitleBarWrapper

    init {
        View.inflate(context, R.layout.view_node_add_layout, this)
        titleBar = TitleBarWrapper(findViewById(R.id.titleBar))

        mEditTextHost = findViewById<EditText>(R.id.edittext_node_host)
        mEditTextPort = findViewById<EditText>(R.id.edittext_node_port)
        mEditTextAccountId = findViewById<EditText>(R.id.edittext_node_account)
        mCheckBoxIsActive = findViewById<CheckBox>(R.id.checkBox_active)
        mButtonsave = findViewById<Button>(R.id.btn_save)


        mButtonsave.setOnClickListener {
            screen?.onSaveClick(mEditTextHost.text.toString(), mEditTextPort.text.toString(),
                    mEditTextAccountId.text.toString(), !mCheckBoxIsActive.isChecked)
        }

        reloadData()
    }


    fun reloadData() {
        titleBar.setTitle(if (params.node == null) "Add Node" else "Edit Node")

        params.node?.let { node ->
            node.host?.let { mEditTextHost.setText(it) }
            mEditTextPort.setText(node.port.toString())
            mEditTextAccountId.setText(node.accountID().stringRepresentation())
            mCheckBoxIsActive.isChecked = !node.disabled
        }

    }
}